modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -165,6 +166,35 @@ public static string NameWithTypeParameters(this INamedTypeSymbol symbol)
         public static string FullQualifiedName(this INamespaceSymbol namespaceSymbol)
             => namespaceSymbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
 
+        public static string FullQualifiedName(this ISymbol symbol)
+            => symbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
+
+        public static string FullQualifiedSyntax(this SyntaxNode node, SemanticModel sm, bool isFirstNode = true)
+        {
+            if (node is NameSyntax ns && isFirstNode)
+            {
+                SymbolInfo nameInfo = sm.GetSymbolInfo(ns);
+                return nameInfo.Symbol?.FullQualifiedName() ?? ns.ToString();
+            }
+
+            StringBuilder sb = new StringBuilder(128);
+            bool innerIsFirstNode = true;
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.HasLeadingTrivia) { sb.Append(child.GetLeadingTrivia()); }
```

Comment:
Our coding style uses new lines even for single lines:

```suggestion
                if (child.HasLeadingTrivia)
                {
                    sb.Append(child.GetLeadingTrivia());
                }
```

Although we sometimes omit the braces, I think that's what @neikeq would prefer. I tend to prefer to always add braces but I don't have a strong opinion.